app/database/permission_granted_store_compute_all_access.go

```diff
@@ -47,63 +47,50 @@ func (s *PermissionGrantedStore) computeAllAccess() {
-				IFNULL(new_data.can_watch_value, 1)
+			IF(MAX(permissions_granted.is_owner), 'transfer', GREATEST(
+				IFNULL(MAX(permissions_granted.can_watch_value), 1),
+				IFNULL(MAX(IF(items_items.watch_propagation, LEAST(parent.can_watch_generated_value, 3 /* answer */), 1)), 1)
 			)) AS can_watch_generated,
-			IF(granted.is_owner, 'transfer', GREATEST(
-				IFNULL(granted.can_edit_value, 1),
-				IFNULL(new_data.can_edit_value, 1)
+			IF(MAX(permissions_granted.is_owner), 'transfer', GREATEST(
+				IFNULL(MAX(permissions_granted.can_edit_value), 1),
+				IFNULL(MAX(IF(items_items.edit_propagation, LEAST(parent.can_edit_generated_value, 3 /* all */), 1)), 1)
 			)) AS can_edit_generated,
-			IFNULL(granted.is_owner, 0) AS is_owner_generated
+			IFNULL(MAX(permissions_granted.is_owner), 0) AS is_owner_generated
 		FROM permissions_propagate
-		LEFT JOIN LATERAL (
-			SELECT
-				MAX(LEAST(
-					IF(LEAST(items_items.descendants_and_solution_view_propagation_value+2, parent.can_view_generated_value) = 3,
-						items_items.content_view_propagation_value,
-						IF(parent.can_view_generated = 'info', 1, parent.can_view_generated_value)),
-					items_items.descendants_and_solution_view_propagation_value+2)) AS can_view_value,
-				MAX(IF(items_items.grant_view_propagation, LEAST(parent.can_grant_view_generated_value, 4), 1)) AS can_grant_view_value,
-				MAX(IF(items_items.watch_propagation, LEAST(parent.can_watch_generated_value, 3), 1)) AS can_watch_value,
-				MAX(IF(items_items.edit_propagation, LEAST(parent.can_edit_generated_value, 3), 1)) AS can_edit_value
-			FROM items_items
-			JOIN permissions_generated AS parent
-				ON parent.item_id = items_items.parent_item_id 
-			JOIN items AS parent_item
-				ON parent_item.id = items_items.parent_item_id
-			WHERE
-				parent.group_id = permissions_propagate.group_id AND items_items.child_item_id = permissions_propagate.item_id
-			GROUP BY permissions_propagate.group_id, permissions_propagate.item_id
-		) AS new_data ON 1
-		LEFT JOIN LATERAL (
-			SELECT
-				MAX(can_view_value) AS can_view_value,
-				MAX(can_grant_view_value) AS can_grant_view_value,
-				MAX(can_watch_value) AS can_watch_value,
-				MAX(can_edit_value) AS can_edit_value,
-				MAX(is_owner) AS is_owner
-			FROM permissions_granted
-			WHERE permissions_granted.group_id = permissions_propagate.group_id AND
-				permissions_granted.item_id = permissions_propagate.item_id
-			GROUP BY permissions_granted.group_id, permissions_granted.item_id
-		) AS granted ON 1
-		WHERE propagate_access = 'self'
+		LEFT JOIN items_items ON items_items.child_item_id = permissions_propagate.item_id
+		LEFT JOIN permissions_generated AS parent
+		  ON parent.item_id = items_items.parent_item_id AND parent.group_id = permissions_propagate.group_id
+		LEFT JOIN permissions_granted
+		  ON permissions_granted.group_id = permissions_propagate.group_id AND
+		    permissions_granted.item_id = permissions_propagate.item_id
```

Comment:
`LEFT JOIN permissions_granted USING (group_id, item_id)`?

Reply:
```Error 1052: Column 'group_id' in from clause is ambiguous```

Reply:
If you move this line two lines above, you'll not have the pb as it is because of the "ON" of the previous join. 
I would prefer this option if you do not mind.

Reply:
fixed